android/MobiledgeXSDKDemo/computervision/src/main/java/com/mobiledgex/computervision/ObjectClassRenderer.java

```diff
@@ -0,0 +1,197 @@
+     * @param imageRect  The coordinates of the TextureView that is showing the preview image.
+     * @param serverToDisplayRatioX  The ratio of the width of the image sent to the server vs.
+     *                               the width of the preview image being displayed.
+     * @param serverToDisplayRatioY  The ratio of the height of the image sent to the server vs.
+     *                               the height of the preview image being displayed.
+     * @param mirrored  Whether the preview image is being mirrored.
+     */
+    public void setDisplayParms(Rect imageRect,
+                                float serverToDisplayRatioX, float serverToDisplayRatioY, boolean mirrored) {
+        mWidth = imageRect.width();
+        mHeight = imageRect.height();
+        mWidthOff = imageRect.left;;
+        mHeightOff = imageRect.top;
+        mServerToDisplayRatioX = serverToDisplayRatioX;
+        mServerToDisplayRatioY = serverToDisplayRatioY;
+        mMirrored = mirrored;
+    }
+
+    /**
+     * Reset the fade-out animation back to the beginning.
+     */
+    public void restartAnimation() {
+        startAnimation(mAlphaAnim);
+    }
+
+    @Override
+    protected void onDraw(Canvas canvas) {
+        Log.d(TAG, "onDraw() "+getWidth()+","+getHeight()+" mObjects="+(mObjects !=null));
+        super.onDraw(canvas);
+
+        if(mObjects == null) {
+            return;
+        }
+
+        Log.d(TAG, "mObjects.length()="+ mObjects.length());
+
+        int totalObjects = mObjects.length();
+        try {
+            for (int i = 0; i < totalObjects; i++) {
+                JSONObject object = mObjects.getJSONObject(i);
+                Log.d(TAG, i + " object=" + object);
+                JSONArray jsonRect = object.getJSONArray("rect");
+                Rect rect = new Rect();
+                rect.left = jsonRect.getInt(0);
+                rect.top = jsonRect.getInt(1);
+                rect.right = jsonRect.getInt(2);
+                rect.bottom = jsonRect.getInt(3);
+                Log.d(TAG, "received rect=" + rect.toShortString());
+
+                rect.left *= mServerToDisplayRatioX;
+                rect.right *= mServerToDisplayRatioX;
+                rect.top *= mServerToDisplayRatioY;
+                rect.bottom *= mServerToDisplayRatioY;
+
+                if (mMirrored) {
+                    Log.d(TAG, "Mirroring!");
+                    // The image that was processed is what the camera sees, but the image we want to
+                    // overlay the rectangle onto is mirrored. So not only do we have to scale it,
+                    // but we have to flip it horizontally.
+                    rect.left = mWidth - rect.left;
+                    rect.right = mWidth - rect.right;
+                    int tmp = rect.left;
+                    rect.left = rect.right;
+                    rect.right = tmp;
```

Comment:
I think we can simplify the mirroring (unless you think they other way is more intuitive):
int tmp = rect.left
rect.left = mWidth - rect.right;
rect.right = mWidth - tmp;